keys/signer_ed25519.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keys
+
+import (
+	"crypto/ed25519"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/coinbase/rosetta-sdk-go/asserter"
+	"github.com/coinbase/rosetta-sdk-go/types"
+)
+
+// SignerEd25519 is initialized from a keypair
+type SignerEd25519 struct {
+	KeyPair KeyPair
+}
+
+// PublicKey returns the PublicKey of the signer
+func (s SignerEd25519) PublicKey() types.PublicKey {
+	return *s.KeyPair.PublicKey
+}
+
+// Signs arbitrary payloads using a KeyPair
+func (s SignerEd25519) Sign(payload types.SigningPayload) (*types.Signature, error) {
```

Comment:
The way we're using pointers or not is inconsistent, in some cases we return pointers and others not, and I can't seem to identify why we're using one vs the other. What's the decision criteria going into this?

In my go experience, typically people just use struct pointers basically everywhere 🤷

Reply:
Basically I tried to use non-pointers as much as I could, which I looked at https://golang.org/pkg/crypto/#Signer and https://pkg.go.dev/golang.org/x/crypto/ed25519?tab=doc for inspiration. Is that correct? My previous version used pointers everywhere, but don't really know what Go convention is here

Reply:
I’ve read a ton of articles advocating for each approach (pointers vs non-pointers).

For consistency with the rest of rosetta-sdk-go, I believe we should use pointers (except for CurveType and SignatureType) for both function inputs and returned values @juliankoh